02-Custom-Login/app/controllers/home_controller.rb

```diff
@@ -0,0 +1,49 @@
+require "auth0"
+
+class HomeController < ApplicationController
+  attr_reader :client
+
+  def auth
+    v2_creds = { client_id: ENV['AUTH0_CLIENT_ID'],
+     token: '',
+     api_version: 2,
+     domain: ENV['AUTH0_DOMAIN'] }
+
+    @client = Auth0Client.new(v2_creds)
+
+    begin
+      if params[:signup]
+        signup
+      else
+        login
```

Comment:
We should call login always, even after signup to avoid redirecting to dashboard, and coming back to / and login through the secured controlled check.